### DISCUSSION TITLE
Need WEBRTC with multiple audio Track support

### DISCUSSION THREAD

**Discussion by jaggi-mohit** [bluenviron/mediamtx#3056] [Questions]
### Question

Hi everyone,
I need a webrtc stream with multiple audio track support, to do so I have published a RTSP stream on the server with multiple audio tracks using FFMPEG and it is working fine on the server end showing 3 tracks (1 video and 2 audio)
but when i play the webrtc url on 8889 it only contains video with single audio track, is there any way to get desired result from command or through changes in mediamtx.yml file?

I have attached the snapshots for your refrence.

![image](https://github.com/bluenviron/mediamtx/assets/64644750/c9506357-7f4b-46ed-8d70-82a551a458d2)

![image](https://github.com/bluenviron/mediamtx/assets/64644750/6c19d428-e5a6-4bb3-a0d2-dc8a53964465)

Thanks in advance !

**Comment by asturl** [comment]
Not a MediaMTX issue in particular:
https://groups.google.com/g/discuss-webrtc/c/vrw44ZGE0gs/m/2YJ6yUEjBgAJ

**Reply by jaggi-mohit** [reply]
Hi, 
Thanks for the reply does this happen only in the case of Chrome or it can support multiple audio tracks on other browsers though I checked on different browsers but the same issue exists.
moreover is there any other solution by which we could achieve webrtc with multiple audio tracks through mediamtx

**Reply by aler9** [reply]
> is there any other solution by which we could achieve webrtc with multiple audio tracks through mediamtx

Not at the moment. While the WebRTC protocol theoretically supports any number of tracks, in practice there are no clients that support more than two tracks at once.

My advice is to use RTSP, that uses the same data transfer mechanism of WebRTC (that are RTP packets) and supports an arbitrary number of tracks.

In the future we will certainly add multitrack support to HLS too.

**Comment by Dennis-A-P-M-C** [comment]
Do you have any plans to enable multi audio track support for WebRTC aggregation?

**Reply by aler9** [reply]
Please read my previous comment. This is not a server limitation but a browser-side limitation. If you want to complain, you can open issues in Chrome and Firefox bug trackers.
